File: 1-DataPreprocessing/DataPreprocessing.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# Data Preprocessing Template

#Importing the libraries
import numpy as np #mathematical tools library
import matplotlib.pyplot as plt #plotting library 
import pandas as pd #importing and managing data sets library

# Import the data set
DataSet = pd.read_csv('Data.csv')
X = DataSet.iloc[:, :-1].values
Y = DataSet.iloc[:, 3].values

# Managing missing data
from sklearn.preprocessing import Imputer
ImputeVals = Imputer(missing_values = 'NaN', strategy = 'mean', axis = 0)
ImputeVals = ImputeVals.fit(X[:, 1:3])
X[:, 1:3] = ImputeVals.transform(X[:, 1:3])

#Encode categorical data as numbers
from sklearn.preprocessing import LabelEncoder, OneHotEncoder

LabelEncoderX = LabelEncoder()
X[:,0] = LabelEncoderX.fit_transform(X[:,0])
CountryEncoder = OneHotEncoder(categorical_features = [0])
X = CountryEncoder.fit_transform(X).toarray()

LabelEncoderY = LabelEncoder()
Y = LabelEncoderY.fit_transform(Y)

#Splitting data into training and test
from sklearn.model_selection import train_test_split
Xtrain, Xtest, Ytrain, Ytest = train_test_split(X, Y, test_size = 0.2, random_state = 0)

#Feature scaling
from sklearn.preprocessing import StandardScaler
ScaleX = StandardScaler()
Xtrain = ScaleX.fit_transform(Xtrain)
Xtest = ScaleX.transform(Xtest)

File: 1-DataPreprocessing/DataPreprocessing.R
# Data Preprocessing Template

#Importing the data set
DataSet = read.csv('Data.csv')

# Managing missing data
DataSet$Age = ifelse(is.na(DataSet$Age),
                     ave(DataSet$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     DataSet$Age)
DataSet$Salary = ifelse(is.na(DataSet$Salary),
                        ave(DataSet$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        DataSet$Salary)

#Encode categorical data into numeric data
DataSet$Country = factor(DataSet$Country, 
                         levels = c('France', 'Spain', 'Germany'),
                         labels = c(1, 2, 3))
DataSet$Purchased = factor(DataSet$Purchased, 
                         levels = c('No', 'Yes'),
                         labels = c(0, 1))

#Split data into training and test
set.seed(123)
Split = sample.split(DataSet$Purchased, SplitRatio = 0.8)
TrainingSet = subset(DataSet, Split == TRUE)
TestSet = subset(DataSet, Split == FALSE)

#Feature scaling
TrainingSet[, 2:3] = scale(TrainingSet[, 2:3])
TestSet[, 2:3] = scale(TestSet[, 2:3])

